src/service/viewer-impl.js

```diff
@@ -253,48 +253,36 @@ export class Viewer {
     // Wait for document to become visible.
     this.docState_.onVisibilityChanged(this.onVisibilityChange_.bind(this));
 
-
-    /**
-     * Creates an error for the case where a channel cannot be established.
-     * @param {!Error|undefined} reason
-     * @return {!Error}
-     */
-    function getChannelError(reason) {
-      if (reason instanceof Error) {
-        reason.message = 'No messaging channel: ' + reason.message;
-        return reason;
-      }
-      return new Error('No messaging channel: ' + reason);
-    }
-
     /**
      * This promise will resolve when communications channel has been
-     * established or timeout in 5 seconds. The timeout is needed to avoid
+     * established or timeout in 20 seconds. The timeout is needed to avoid
      * this promise becoming a memory leak with accumulating undelivered
-     * messages.
-     * @private @const {!Promise<!Viewer>}
+     * messages. The promise is only available when the document is embedded.
+     * @private @const {?Promise}
      */
-    this.messagingReadyPromise_ = timer.timeoutPromise(
-        20000,
-        new Promise(resolve => {
-          /** @private @const {function(!Viewer)} */
-          this.messagingReadyResolver_ = resolve;
-        })).catch(reason => {
-          throw getChannelError(reason);
-        });
+    this.messagingReadyPromise_ = this.isEmbedded_ ?;
+        timer.timeoutPromise(
+            20000,
+            new Promise(resolve => {
+              /** @private @const {function()|undefined} */
+              this.messagingReadyResolver_ = resolve;
+            })).catch(reason => {
+              throw getChannelError(reason);
+            }) : null;
```

Comment:
Would it be better to return a rejected promise? That'll avoid all the conditionals below.